core/client/utils/createLayers.js

```diff
@@ -41,6 +41,7 @@ export async function createLayersFromDataAssets(
             },
           }),
         },
+        style,
```

Comment:
this would break if the styles include variable expressions. You can try checking if `style.variables` exist first, something like:
```js
 ...(!style?.variables && {style}),
```

Reply:
If i am interpreting this right it would break for vector data with variables, but not for geotiff rendering where variables is already integrated.
I think for geotiff data we use variables most of the time?
So the question is, do we want to support geotiff rendering with variables with this fix, if not we could apply the check proposed by Ahmed

Reply:
Thanks! I have adapted the code according to the feedback.